src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTestBase.cs

```diff
@@ -68,6 +68,16 @@ protected void StartAndKillProcessWithDelay(Process p)
             Assert.True(p.WaitForExit(WaitInMS));
         }
 
+        protected void EnsureState(StateID state)
+        {
+            switch (state)
+            {
+                case StateID.HasLoadedModules:
+                    while (_process.Modules.Count == 0) ;
+                    break;
+            }
+        }
+
```

Comment:
Forgetting for a moment the actual mechanism by which this is checking to ensure that modules are loaded, I'm not understanding why all of this StateID stuff is needed.  Why isn't this method just:
```C#
protected void EnsureModulesLoaded()
{
    ... // do whatever checks and loading we need for modules
}
```
?

That said, I'm not understanding the ```while (_process.Modules.Count == 0);``` part of the solution.  Won't this fail, for example, with the OS X implementation that returns an empty array?  And even if this did work, is there not a better mechanism available, or should we at least do some sort of safer spin wait in order to avoid this kind of tight loop?